src/Umbraco.Core/Services/Implement/MemberGroupService.cs

```diff
@@ -70,6 +70,14 @@ public IMemberGroup GetById(int id)
             }
         }
 
+        public IMemberGroup GetById(Guid id)
+        {
+            using (var scope = ScopeProvider.CreateScope(autoComplete: true))
+            {
+                return _memberGroupRepository.GetMany().FirstOrDefault(x => x.Key == id);
```

Comment:
@bjarnef I seem to have missed this line! I realize you had no other choice here, but this is not a friendly with regards to performance! It will get ALL the row from SQL and do filtering in C#, that can potentially be many rows and this repository doesn't cache the results I think.

Can you please look into adding a method to the `MemberGroupRepository` to do a single SQL query to get a group by it's `Key`?

Reply:
Ahh yes. Need to have another look at this. I guess for most projects it won't be a huge performance issues, since no or a few member groups are used, but some projects might have a huge among of member groups.

Found this for media:
https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Core/Services/Implement/MediaService.cs#L346-L358

Reply:
Sure, I am more concerned about people using that method in all of their views so it will run on each request! Something like MediaService or ContentService would be good yep!

Reply:
I have submitted a PR here: https://github.com/umbraco/Umbraco-CMS/pull/8496